galasa-extensions-parent/dev.galasa.ecosystem.builder/src/main/java/galasa/ecosystem/docker/builder/DockerEcosystemBuilder.java

```diff
@@ -0,0 +1,628 @@
+			        break;
+			    } else {
+			        try {
+			        	String port1 = null;
+			        	String port2 = null;
+			        	port1 = resPort.substring(0, resPort.indexOf(":"));
+			        	port2 = resPort.substring(resPort.lastIndexOf(":") + 1);
+			        	Integer.valueOf(port1);
+			            Integer.valueOf(port2);
+			            break;
+			        } catch (NumberFormatException | StringIndexOutOfBoundsException ex) {
+			            System.out.println("Please enter a port in the format nnnn:nnnn");
+			        }
+			    }
+			}
+		  System.out.println("Setting port to: " + resPort + "\n");
+		  
+		  System.out.println("Moving on to SimBank container setup..." + "\n");
+		  
+		  System.out.println("SimBank container name: (default galasa): ");
+		  sbName = scanner.nextLine();
+		  if (sbName.isEmpty() || sbName.equals("galasa")) {
+			  sbName = "galasa";
+			  System.out.println("Using default container name...");
+		  }
+		  else {
+			  System.out.println("Using new container name...");
+		  }
+		  
+		  System.out.println("SimBank container name set to: " + sbName + "\n");
+		  
+		  System.out.println("Using previously set network name: " + netName + "\n");
+		  
+		  System.out.println("Restart policy - no, on-failure, always or unless-stopped (default always): ");
+		  sbResPol = scanner.nextLine();
+		  while (!(sbResPol.toUpperCase().equals("NO") || sbResPol.toUpperCase().equals("ON-FAILURE") || sbResPol.toUpperCase().equals("ALWAYS") || sbResPol.toUpperCase().equals("UNLESS-STOPPED") || sbResPol.isEmpty())) {	
+		      System.out.println("I'm sorry but that isn't a valid input, please ensure you have entered a restart policy listed above");
+		      sbResPol = scanner.nextLine();
+			}
+		  if (sbResPol.isEmpty() || sbResPol.equals("always")) {
+			  sbResPol = "always";
+			  System.out.println("Using default restart policy...");
+		  }
+		  else {
+			  System.out.println("Using new restart policy");
+		  }
+		  
+		  System.out.println("Restart policy set to: " + sbResPol + "\n");
+		  
+		  System.out.println("Port to publish to host: (default 2080:2080)");
+		  while (true) {
+			    sbPort = scanner.nextLine();
+			    if (sbPort.isEmpty() || sbPort.equals("2080:2080")) {
+			        sbPort = "2080:2080";
+			        break;
+			    } else {
+			        try {
+			        	String port1 = null;
+			        	String port2 = null;
+			        	port1 = sbPort.substring(0, sbPort.indexOf(":"));
+			        	port2 = sbPort.substring(sbPort.lastIndexOf(":") + 1);
+			        	Integer.valueOf(port1);
+			            Integer.valueOf(port2);
+			            break;
+			        } catch (NumberFormatException | StringIndexOutOfBoundsException ex) {
+			            System.out.println("Please enter a port in the format nnnn:nnnn");
+			        }
+			    }
+			}
+		  
+		  System.out.println("Setting port to: " + sbPort + "\n");
+		  
+		  System.out.println("Docker config set up complete..." + "\n");
+		  
```

Comment:
At this point you should present an option of all the settings and ask them if it is ok
if not, give them the option to change which bit and come back here